sdk/eventhubs/azure-messaging-eventhubs/inc/azure/messaging/eventhubs/event_data_batch.hpp

```diff
@@ -127,15 +128,15 @@ namespace Azure { namespace Messaging { namespace EventHubs {
      *
      * @returns true if the message was added to the batch, false otherwise.
      */
-    bool TryAddMessage(Azure::Messaging::EventHubs::Models::EventData const& message);
+    _azure_NODISCARD bool TryAdd(Azure::Messaging::EventHubs::Models::EventData const& message);
```

Comment:
why rename this ? , there is also the TryAddAmqpMessage , maybe we should look into the names matching, i know that one is private ,

Reply:
Because the things you add aren't always messages. Sometimes they're an `Azure::Core::Amqp::Models::AmqpMessage`, sometimes they are an `EventData`.

Because they're not always a Message, it doesn't make sense to put Message in the method name.

`TryAddAmqpMessage` is the private worker function which does the actual add operation, and it always accepts an `Azure::Core::Amqp::Models::AmqpMessage`.